torch/_dynamo/convert_frame.py

```diff
@@ -508,6 +518,24 @@ def __call__(
         if is_generator(code):
             unimplemented("generator")
 
+        f_locals = frame.f_locals
+        if (
+            frame.f_code.co_name in ("__enter__", "__exit__")
+            and len(f_locals) == 1
```

Comment:
`__exit__` has 4 arguments, right? does it need a different len(f_locals) check?

Reply:
what causes `__exit__` to always be skipped?

Reply:
There's no tensor in the frame.

Reply:
is it possible for the user to put a tensor in the exit function?

Reply:
I don't think so. The `__exit__` function is implemented in CPython [contextlib.py](https://github.com/python/cpython/blob/e8dbe7ec573c50d082d4b7a32fe54ead0937557a/Lib/contextlib.py#L145-L200)

Reply:
Actually, maybe if one raises an exception with a tensor inside, the `__exit__` frame will have a tensor:

```python
raise ValueError(a_tensor)`.
```

Reply:
Update on this one: I had to comment a few checks to get this code "working" inside context manager. Thus, I don't think it is possible for a torch.compile user to run this code.